src/msix/PAL/FileSystem/Win32/DirectoryObject.cpp

```diff
@@ -165,6 +173,31 @@ namespace MSIX {
         auto result = ComPtr<IStream>::Make<FileStream>(std::move(utf8_to_wstring(path)), mode);
         return result;
     }
+
+    std::multimap<std::uint64_t, std::string> DirectoryObject::GetFilesByLastModDate()
+    {
+    #ifdef MSIX_PACK
+        std::multimap<std::uint64_t, std::string> files;
+        WalkDirectory<WalkOptions::Recursive | WalkOptions::Files>(m_root, [&](
+                std::string root,
+                WalkOptions option,
+                std::string&& name,
+                std::uint64_t size)
+            {
+                if (name != "AppxManifest.xml") // should only add payload files to the map
+                {
+                    std::string fileName = root + GetPathSeparator() + name;
+                    // root contains the top level directory, which we don't need
+                    fileName = fileName.substr(fileName.find_first_of(GetPathSeparator()) + 1);
+                    files.insert(std::make_pair(size, std::move(fileName)));
+                }
+                return true;
+            });
+        return files;
+    #else
+        NOTIMPLEMENTED;
+    #endif // MSIX_PACK

```

Comment:
I suspect we may see a few of these, not sure if there is a better way.

Reply:
better = more concise

---
In reply to: [276759188](https://github.com/Microsoft/msix-packaging/pull/93#discussion_r276759188) [](ancestors = 276759188)

Reply:
#define MSIX_PACK_ONLY
#define MSIX_PACK_ONLY_END

If MSIX_PACK => both empty
Else only => /*
end => */ NOTIMPLEMENTED

MSIX_PACK_ONLY
<code>
MSIX_PACK_ONLY_END

Thoughts?  Prevents use of multiline comments within the code, but I don't think they should be used in normal course anyway.

---
In reply to: [276759833](https://github.com/Microsoft/msix-packaging/pull/93#discussion_r276759833) [](ancestors = 276759833,276759188)

Reply:
This won't work. When ONLY is preprocessed and add /* it will comment out  the rest of the file and fail with "C1071: unexpected end of file found in comment" because the second macro is now part of the comment.

I end up doing:
if not MSIX_PACK
#define MSIX_PACK_ONLY throw 

and also apply it to bundle support

---
In reply to: [276767727](https://github.com/Microsoft/msix-packaging/pull/93#discussion_r276767727) [](ancestors = 276767727,276759833,276759188)